### ISSUE TITLE
golem.client check_payments failed - an exception is raised.

### ISSUE THREAD

**Issue by ggruszczynski** [golemfactory/clay#1605]
I let run Golem for 2 days and...
golem.client  check_payments failed - an exception is raised

Probably connected with:

- Requestor who does not pay should be banned #1604
- golem is multiplying one transaction several times in history of payments tab #1521

TODO:
How long to store expected_income before banning Requestor?

Additionally, it seems  that `Trust.PAYMENT.decrease(node_id)` is never called

```python
    def check_payments(self):
        if not self.transaction_system:
            return
        after_deadline_nodes = self.transaction_system.check_payments()
        for node_id in after_deadline_nodes:
            Trust.PAYMENT.decrease(node_id)
```

log
```
2017-11-01 03:54:16 ERROR    golem.client                        check_payments failed
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/peewee.py", line 3181, in get
    return next(clone.execute())
  File "/usr/local/lib/python3.5/dist-packages/peewee.py", line 2346, in next
    obj = self.iterate()
  File "/usr/local/lib/python3.5/dist-packages/peewee.py", line 2328, in iterate
    raise StopIteration
StopIteration

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/ggruszczynski/PycharmProjects/golem/golem/transactions/incomeskeeper.py", line 36, in run_once
    subtask=expected_income.subtask,
  File "/usr/local/lib/python3.5/dist-packages/peewee.py", line 4926, in get
    return sq.get()
  File "/usr/local/lib/python3.5/dist-packages/peewee.py", line 3185, in get
    % self.sql())
golem.model.IncomeDoesNotExist: Instance matching query does not exist:
SQL: SELECT "t1"."created_date", "t1"."modified_date", "t1"."sender_node", "t1"."task", "t1"."subtask", "t1"."transaction", "t1"."block_number", "t1"."value" FROM "income" AS t1 WHERE ((("t1"."sender_node" = ?) AND ("t1"."subtask" = ?)) AND ("t1"."task" = ?))
PARAMS: ['e5b5aa48b886666696f006daeb9b27f5cfa600bf8f50c4b5fff95703fdfd13a011d13e93c548d3bf367f8c305a0b91b3a0bb09a8f7876dcf31208dd048b1abf6', '276049019867471894227184876958256397136', '16e20f0b-4439-4f71-a0a5-8f8dd77e7be9']

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/peewee.py", line 3768, in execute_sql
    cursor.execute(sql, params or ())
sqlite3.OperationalError: database is locked

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/ggruszczynski/PycharmProjects/golem/golem/client.py", line 1007, in __do_work
    self.check_payments()
  File "/home/ggruszczynski/PycharmProjects/golem/golem/client.py", line 919, in check_payments
    after_deadline_nodes = self.transaction_system.check_payments()
  File "/home/ggruszczynski/PycharmProjects/golem/golem/transactions/transactionsystem.py", line 96, in check_payments
    self.incomes_keeper.run_once()
  File "/home/ggruszczynski/PycharmProjects/golem/golem/transactions/incomeskeeper.py", line 42, in run_once
    expected_income.save()
  File "/usr/local/lib/python3.5/dist-packages/peewee.py", line 5103, in save
    rows = self.update(**field_dict).where(self._pk_expr()).execute()
  File "/usr/local/lib/python3.5/dist-packages/peewee.py", line 3400, in execute
    return self.database.rows_affected(self._execute())
  File "/usr/local/lib/python3.5/dist-packages/peewee.py", line 2912, in _execute
    return self.database.execute_sql(sql, params, self.require_commit)
  File "/usr/local/lib/python3.5/dist-packages/peewee.py", line 3772, in execute_sql
    raise
  File "/usr/local/lib/python3.5/dist-packages/peewee.py", line 3598, in __exit__
    reraise(new_type, new_type(*exc_args), traceback)
  File "/usr/local/lib/python3.5/dist-packages/peewee.py", line 135, in reraise
    raise value.with_traceback(tb)
  File "/usr/local/lib/python3.5/dist-packages/peewee.py", line 3768, in execute_sql
    cursor.execute(sql, params or ())
peewee.OperationalError: database is locked
```

The `expected_income.delete_instance()` is called only if Requestor pays (there is an Income):

![image](https://user-images.githubusercontent.com/6128943/32320698-ee08c9be-bfbe-11e7-82e8-f0f6ddabac9a.png)

**Comment by ggruszczynski** [comment]
same as https://github.com/golemfactory/golem/issues/1600

Update: Actually slightly different, there is: `peewee.OperationalError: unable to open database file`

**Comment by ggruszczynski** [comment]
Update: the `expected_income.delete_instance()` is actually called only when Requestor pays (there is an Income)

**Comment by Wiezzel** [comment]
This error no longer occurs.
`Trust.PAYMENT.decrease()` will be addressed as a separate issue: #2267
